### DISCUSSION TITLE
Is there a reason why shouldInterceptRequest is not implemented for Android

### DISCUSSION THREAD

**Discussion by AbrahamOEd** [react-native-webview/react-native-webview#3402] [Q&A]
Wondering if there were previous attempts at implementation this functionality for Android. Would really appreciate if there are any known blockers for this functionality to be implemented. Thanks

**Comment by AbrahamOEd** [comment]
Asking the above as I am looking to try and implement it myself. A current blocker that I'm facing is how we could provide the `InputStream` from React Native to Android native (for audio, files, etc.). Otherwise, the implementation may only be able to handle String/ByteArrayInputStreams

**Comment by AbrahamOsmondE** [comment]
https://github.com/react-native-webview/react-native-webview/pull/3422, here's my first attempt at implementing the function. It still have some hardcoded values, but works as a PoC
